plugins/explore/src/plugin.cpp

```diff
@@ -0,0 +1,244 @@
+public:
+  explore_operator() = default;
+
+  explicit explore_operator(operator_args args) : args_{std::move(args)} {
+  }
+
+  auto name() const -> std::string override {
+    return "explore";
+  }
+
+  auto
+  operator()(generator<table_slice> input, operator_control_plane& ctrl) const
+    -> generator<std::monostate> {
+    if (!ctrl.has_terminal()) {
+      diagnostic::error("no terminal found")
+        .note("{} operator requires terminal", name())
+        .emit(ctrl.diagnostics());
+      co_return;
+    }
+    using namespace ftxui;
+    auto screen = make_interactive_screen(args_);
+    auto state = make_ui_state(args_);
+    // Ban UI main loop into dedicated thread.
+    auto thread = std::thread([&] {
+      auto main = MainWindow(&screen, &state);
+      screen.Loop(main);
+    });
+    for (auto&& slice : input) {
+      if (slice.rows() == 0) {
+        co_yield {};
+        continue;
+      }
+      // The task executes inside the UI thread. Therefore state access is
+      // thread-safe.
+      auto task = [&state, &screen, slice] {
+        auto& table = state.tables[slice.schema()];
+        if (!table)
+          table = std::make_shared<table_state>();
+        table->slices.push_back(slice);
+        screen.PostEvent(Event::Custom); // Redraw screen
+      };
+      screen.Post(task);
+      co_yield {};
+    }
+    thread.join();
+  }
+
+  friend auto inspect(auto& f, explore_operator& x) -> bool {
+    (void)f, (void)x;
+    return true;
+  }
+
+  auto location() const -> operator_location override {
+    return operator_location::local;
+  }
+
+private:
+  operator_args args_;
+};
+
+class table_printer final : public plugin_printer {
+public:
+  table_printer() = default;
+
+  explicit table_printer(printer_args args) : args_{std::move(args)} {
+  }
+
+  // FIXME: this should actually be "table", but it's currently not possible.
+  auto name() const -> std::string override {
+    return "explore";
+  }
+
+  auto instantiate(type, operator_control_plane&) const
+    -> caf::expected<std::unique_ptr<printer_instance>> override {
+    class screen_printer : public printer_instance {
+    public:
+      screen_printer(const printer_args args)
+        : args_{std::move(args)}, state_{make_ui_state(args_)} {
+      }
+
+      auto process(table_slice slice) -> generator<chunk_ptr> override {
+        if (slice.rows() == 0) {
+          co_yield {};
+          co_return;
+        }
+        auto& table = state_.tables[slice.schema()];
+        if (!table)
+          table = std::make_shared<table_state>();
+        table->slices.push_back(slice);
+        auto& component = components_[slice.schema()];
+        if (!component)
+          component = DataFrame(&state_, slice.schema());
+        if (!args_.real_time)
+          co_return;
+        auto result = chunk::make(to_string(component));
+        state_.tables.clear();
+        components_.clear();
+        co_yield result;
```

Comment:
This does not handle narrow screens well. Anything we can do about that?

<img width="1840" alt="image" src="https://github.com/tenzir/tenzir/assets/4488655/2fb2ff9c-2dd7-4290-8046-c98b3b337e80">

Reply:
I'm not sure yet how to fix this. I think this is the same effect I'm describing at https://github.com/ArthurSonzogni/FTXUI/discussions/681#discussioncomment-6519431. @ArthurSonzogni perhaps the solution to this is a no-brainer for you when you see this screenshot?

Reply:
Noted down as follow-up.